### DISCUSSION TITLE
imbalanced multi-label dataset

### DISCUSSION THREAD

**Discussion by georgemossad** [timeseriesAI/tsai#740] [Q&A] [answered]
hey everyone,
i am trying to use the get_split function on my multi-labeled dataset which is imbalanced as it has minor classes and also major classes. but when setting balance = True in the get_split function i get the following:

'' ValueError: You appear to be using a legacy multi-label data representation. The sequence of sequences are no longer supported; use a binary array or sparse matrix instead - the MultiLabelBinarizer transformer can convert to this format. ''

 so is there a way to deal with this problem using this library?
 thanks

**Comment by oguiza** [comment]
For multi-labeled datasets, I don't what's the best way to handle it.
Anybody else has an idea?

**Reply by georgemossad** [reply]
@oguiza is there at least a way to deal with imbalanced multi-labeled during training by assigning weights or something similar?

I saw this solution in the other discussion and used it so does it work:

''
loss_func=BCEWithLogitsLossFlat(pos_weight=dls.train.cws.sqrt())
 ''

**Reply by oguiza** [reply, marked as answer]
Yes, you can use weights in 2 ways:

- Pass class weights to the loss function 
- Pass a weight by sample when building the dls. the parameter is called weights. You can calculate weights any way you want.

**Reply by georgemossad** [reply]
@oguiza Great thanks for your time!

**Comment by aying0818** [comment]
Hello, 

i am trying to use the BCEWithLogitsLossFlat(pos_weight=dls.train.cws) in my dataset(2 classes). I am getting the following error after passing the class weights to the loss function:

**ValueError: Target size (torch.Size([128])) must be the same as input size (torch.Size([128, 2]))**

The error code is: 
**learn = Learner(dls, model, metrics=accuracy, opt_func=Adam, loss_func=BCEWithLogitsLossFlat(pos_weight=dls.train.cws))**
**learn.lr_find()**

Notably, the same error was reported when I tried to reproduce tutorial file '01a_Multi-class classification'. This is a 6-categorization problem.

**ValueError: Target size (torch.Size([384])) must be the same as input size (torch.Size([2304]))**
The error code is: 
**learn = ts_learner(dls, metrics=accuracy) # == ts_learner(dls, arch=InceptionTimePlus, metrics=accuracy) since InceptionTimePlus is the default arch**
**learn.lr_find()**

I would appreciate your help in checking what is causing this. Looking forward to your reply.
